src/scanpy/preprocessing/_pca_dask_sparse.py

```diff
@@ -0,0 +1,195 @@
+from __future__ import annotations
+
+import math
+
+import cupy as cp
+from cuml.internals.memory_utils import with_cupy_rmm
+from rapids_singlecell._compat import _meta_dense
+from rapids_singlecell.preprocessing._utils import _get_mean_var
+
+
+class PCA_sparse_dask:
+    def __init__(self, n_components) -> None:
+        self.n_components = n_components
+
+    def fit(self, x):
+        if self.n_components is None:
+            n_rows = x.shape[0]
+            n_cols = x.shape[1]
+            self.n_components_ = min(n_rows, n_cols)
+        else:
+            self.n_components_ = self.n_components
+
+        self.n_samples_ = x.shape[0]
+        self.n_features_in_ = x.shape[1] if x.ndim == 2 else 1
+        self.dtype = x.dtype
+        covariance, self.mean_, _ = _cov_sparse_dask(x=x, return_mean=True)
+        self.explained_variance_, self.components_ = cp.linalg.eigh(
+            covariance, UPLO="U"
+        )
```

Comment:
this part doesn’t need to be in dask, `_cov_sparse_dask` and `transform` have to